File: java/src/test/java/problems1/StringToIntegerAtoiTest.java
package problems1;

import org.junit.Test;
import testing.Sample;
import testing.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class StringToIntegerAtoiTest extends SampledTest {
    @Override
    public String getFileName() {
        return "string_to_integer_atoi.json";
    }

    @Test
    public void test() {
        for (StringToIntegerAtoi solution : StringToIntegerAtoi.values()) {
            for (TypedSample sample : getSamples(TypedSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.myAtoi(sample.input.s))
                    .isEqualTo(sample.output);
            }
        }
    }

    public static class TypedSample extends Sample<TypedSample.Input, Integer> {
        public static class Input {
            public String s;
        }
    }
}

File: typescript/__tests__/problems1_100/string-to-integer-atoi.test.ts
import {strictEqual} from 'assert';
import SOLUTIONS from '../../src/problems1_100/string-to-integer-atoi';
import getSamples from '../../../testing/js/src/sample/samples';

type Sample = {
  input: {
    s: string,
  },
  output: number,
  message: string,
}

describe(
    'StringToIntegerAtoi',
    () =>
        test(
            'test',
            () =>
                SOLUTIONS.forEach(solution =>
                    getSamples().forEach((sample: Sample) =>
                        strictEqual(
                            solution.myAtoi(sample.input.s),
                            sample.output,
                            sample.message,
                        ),
                    ),
                ),
        ),
);
